src/Entity/ShipmentTotalWeight.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Ups\Entity;
+
+use DOMDocument;
+use DOMElement;
+use Ups\NodeInterface;
+use Ups\Entity\UnitOfMeasurement;
+
+class ShipmentTotalWeight implements NodeInterface
+{
+    private $unitOfMeasurement;
+
+    private $Weight;
+
+    public function __construct($response = null)
+    {
+        if (null !== $response) {
+            if (isset($response->unitOfMeasurement)) {
+                $this->setUnitOfMeasurement(new UnitOfMeasurement($response->unitOfMeasurement));
+            }
+            if (isset($response->Weight)) {
+                $this->setWeight($response->Weight);
+            }
+        }
+    }
+
+    /**
+     * @param null|DOMDocument $document
+     *
+     * @return DOMElement
+     */
+    public function toNode(DOMDocument $document = null)
+    {
+        if (null === $document) {
+            $document = new DOMDocument();
+        }
+
+        $node = $document->createElement('ShipmentTotalWeight');
+        if ($this->getUnitOfMeasurement()) {
+            $node->appendChild($this->getUnitOfMeasurement()->toNode($document));
+        }
+
+        $node->appendChild($document->createElement('Weight', $this->getWeight()));
+        
+        return $node;
+    }
+
+    /**
+     * @return UnitOfMeasurement
+     */
+    public function getUnitOfMeasurement()
+    {
+        return $this->unitOfMeasurement;
+    }
+
+
+    /**
+     * @param UnitOfMeasurement $unitOfMeasurement
+     */
+    public function setUnitOfMeasurement(UnitOfMeasurement $unitOfMeasurement)
+    {
+        $this->unitOfMeasurement = $unitOfMeasurement;
+    }
+
+    /**
+     * @return string weight
+     */
+    public function getWeight()
+    {
+        return $this->Weight;
+    }
+    
+    /**
+     * @param number $weight
```

Comment:
`number` should be either `int` or `string` and same hint used accordingly in docblock / return hint